python-sdk/src/client.rs

```diff
@@ -0,0 +1,709 @@
+            None
+        };
+
+        Ok(EvaluationResult {
+            variation,
+            action,
+            evaluation_details,
+        })
+    }
+}
+
+#[pyclass(frozen, module = "eppo_client")]
+pub struct EppoClient {
+    configuration_store: Arc<ConfigurationStore>,
+    poller_thread: Option<PollerThread>,
+    assignment_logger: Py<AssignmentLogger>,
+    is_graceful_mode: AtomicBool,
+}
+
+#[pymethods]
+impl EppoClient {
+    fn get_string_assignment(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: Py<PyString>,
+    ) -> PyResult<PyObject> {
+        slf.get().get_assignment(
+            slf.py(),
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::String),
+            default.into_any(),
+        )
+    }
+    fn get_integer_assignment(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: Py<PyInt>,
+    ) -> PyResult<PyObject> {
+        slf.get().get_assignment(
+            slf.py(),
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Integer),
+            default.into_any(),
+        )
+    }
+    fn get_numeric_assignment(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: Py<PyFloat>,
+    ) -> PyResult<PyObject> {
+        slf.get().get_assignment(
+            slf.py(),
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Numeric),
+            default.into_any(),
+        )
+    }
+    fn get_boolean_assignment(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: Py<PyBool>,
+    ) -> PyResult<PyObject> {
+        slf.get().get_assignment(
+            slf.py(),
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Boolean),
+            default.into_any(),
+        )
+    }
+    fn get_json_assignment(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: PyObject,
+    ) -> PyResult<PyObject> {
+        slf.get().get_assignment(
+            slf.py(),
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Json),
+            default.into_any(),
+        )
+    }
+
+    fn get_string_assignment_details(
+        slf: &Bound<EppoClient>,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: Attributes,
+        default: Py<PyString>,
+    ) -> PyResult<EvaluationResult> {
+        slf.get().get_assignment_details(
```

Comment:
would it make sense for the underlying rust function to have details passed in as boolean like pass in type? Then one less "public" function upstream SDKs need to worry about.

Reply:
Do you mean a type-level parameter like this?
```rs
get_assignment::<WithDetails>(...)
```
It's possible but is a bit convoluted to implement (requires setting up traits and proxying implementation through trait), so isn't worth it for internal function